### PR TITLE
Properly handle game pause in VideoPlayer

### LINKED ISSUE #33702: After unpause game, VideoPlayer freezes; everything else unpauses
Godot version: 3.1

OS/device including version: Windows 10 Pro

Issue description: After setting `get_tree()pause = false`, VideoPlayer does not continue playing video, but freezes while other game mechanics continue. I have the VideoPlayer looping a background video as environment during gameplay.

Steps to reproduce: 
1. Create VideoPlayer node and load in any video file. 
2. Create any Button to pause and unpause with code `get_tree()pause = true`,  `get_tree()pause = false`
3. Run game and test.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #33702

### CODE REVIEW

File: scene/gui/video_player.cpp [hunk 1]
```diff
@@ -186,6 +186,18 @@ void VideoPlayer::_notification(int p_notification) {
...
+
+		case NOTIFICATION_PAUSED: {
+			playback->set_paused(true);
+			set_process_internal(false);
+			last_audio_time = 0;
+		} break;
+
+		case NOTIFICATION_UNPAUSED: {
+			playback->set_paused(false);
+			set_process_internal(true);
```
Comment (Xrayez) [inline scene/gui/video_player.cpp:198]: Does it mean that the playback is going to be resumed even if a video was paused manually (`$video.paused = true`) by a user? (not via `get_tree().paused`).

Comment (YeldhamDev) [reply scene/gui/video_player.cpp:198]: Xrayez is right, it should differentiate if it was paused manually or not before unpausing it.

**Comment by KoBeWi** [comment]
Rebased and reworked. I added `paused_from_tree` variable, which is set to true when the player is playing and gets paused from scene tree. I couldn't test if it works though, because VideoStreamPlayer doesn't seem functional, idk

File: scene/gui/video_stream_player.cpp [hunk 2]
```diff
@@ -177,8 +177,30 @@ void VideoStreamPlayer::_notification(int p_notification) {
...
-	};
-};
+
+		case NOTIFICATION_PAUSED: {
+			if (is_playing() && !is_paused()) {
+				paused_from_tree = true;
+				if (playback.is_valid()) {
+					playback->set_paused(true);
+					set_process_internal(false);
+				};
```
Comment (akien-mga) [inline scene/gui/video_stream_player.cpp:187]: Trailing semicolon (also in `UNPAUSED`).

**Comment by KoBeWi** [comment]
Tested and it works correctly.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**